File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/SearchContextInstancesExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class SearchContextInstancesExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("ApiKey")).setApiKey("YOUR_API_KEY");

        var contextInstanceSearch = new ContextInstanceSearch();
        contextInstanceSearch.filter("{\"filter\": \"kindKeys:{\"contains\": [\"user:Henry\"]},\"sort\": \"-ts\",\"limit\": 50}");
        contextInstanceSearch.sort("-ts");
        contextInstanceSearch.limit(10);
        contextInstanceSearch.continuationToken("QAGFKH1313KUGI2351");

        try
        {
            var response = new ContextsApi(config).searchContextInstances(
                null, // projectKey
                null, // environmentKey
                contextInstanceSearch,
                null, // limit
                null, // continuationToken
                null, // sort
                null, // filter
                null // includeTotalCount
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling ContextsApi#searchContextInstances");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/SearchContextInstancesExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class SearchContextInstancesExample
{
    fun searchContextInstances()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val contextInstanceSearch = ContextInstanceSearch(
            filter = "{\"filter\": \"kindKeys:{\"contains\": [\"user:Henry\"]},\"sort\": \"-ts\",\"limit\": 50}",
            sort = "-ts",
            limit = 10,
            continuationToken = "QAGFKH1313KUGI2351",
        )

        try
        {
            val response = ContextsApi().searchContextInstances(
                projectKey = "projectKey_string",
                environmentKey = "environmentKey_string",
                contextInstanceSearch = contextInstanceSearch,
                limit = null,
                continuationToken = null,
                sort = null,
                filter = null,
                includeTotalCount = null,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling ContextsApi#searchContextInstances")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling ContextsApi#searchContextInstances")
            e.printStackTrace()
        }
    }
}
